internal/clients/telegram/state/daily_status.go

```diff
@@ -0,0 +1,235 @@
+	return s.handleDailyStatus(message.Chat.ID, message.Text)
+}
+
+func (s *DailyStatusHandler) CallbackQuery(callback update.CallbackQuery) Transition {
+	ignoreWithAlert := NewTransition(s.RootState, s.userData, []response.BotAction{
+		response.CallbackQueryAnswerNotification(callback, "This button doesnt work."),
+	})
+
+	data, isSome := callback.Data.Unwrap()
+	if !isSome {
+		return ignoreWithAlert
+	}
+
+	message, isSome := callback.Message.Unwrap()
+	if !isSome {
+		return ignoreWithAlert
+	}
+
+	if transition, handled := s.handleCQSaveDefaultProject(data, message); handled {
+		return transition
+	}
+
+	return ignoreWithAlert
+}
+
+func (s *DailyStatusHandler) Ignore() Transition {
+	return NewTransition(s.DailyStatusState, s.userData, response.Nothing())
+}
+
+//nolint:cyclop
+func (s *DailyStatusHandler) handleDailyStatus(chatID update.ChatID, text string) Transition {
+	cmd, isCmd := slashcmd.Parse(text)
+
+	if isCmd && strings.ToLower(cmd.Method) == cancelCommand {
+		return NewTransition(s.RootState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)), "Canceled."),
+		})
+	}
+
+	switch s.Stage {
+	case ignoreMessagesDailyStatusStage:
+		break
+
+	case discoveryOfTheDayDailyStatusStage:
+		s.DailyStatusState.Stage = questionsAndBlockersDailyStatusStage
+
+		if isCmd && strings.ToLower(cmd.Method) == noneCommand {
+			s.DiscoveryOfTheDay = option.None[string]()
+		} else {
+			s.DiscoveryOfTheDay = option.Some(text)
+		}
+
+		return NewTransition(s.DailyStatusState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)),
+				s.responses.QuestionsAndBlockers,
+			),
+		})
+
+	case questionsAndBlockersDailyStatusStage:
+		if isCmd && strings.ToLower(cmd.Method) == noneCommand {
+			s.QuestionsAndBlockers = option.None[string]()
+		} else {
+			s.QuestionsAndBlockers = option.Some(text)
+		}
+
+		report, err := s.generateReport()
+		if err != nil {
+			report = github.GqlErrorStringOr("Github API error: %s", err, "Something went wrong while contacting GitHub.")
+		}
+
+		return NewTransition(s.RootState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)), report),
+		})
+	}
+
+	return s.Ignore()
+}
+
+/*
+Checks if the query is to save the default project. The retuned bool indicates if the query was about saving the default
+project. If returns false try other queries if present or return an ignore response.
+*/
+func (s *DailyStatusHandler) handleCQSaveDefaultProject(data string, message update.Message) (Transition, bool) {
+	askDiscoveryOfTheDay := []response.BotAction{
+		response.RemoveReplyMarkup(message),
+		response.NewSendMessage(response.ChatID(fmt.Sprint(message.Chat.ID)), s.responses.DiscoveryOfTheDay),
+	}
+
+	if data == cqDailyStatusAskDefaultProjectEveryTime {
+		s.UseOnlyProjectNoSaveDefault = true
```

Comment:
This boolean is named and used completely not how it is intended. I think its best to remove it and in root state if the user has 1 project send a message saying that project is saved as the default project and immediately save it into default project field on the state. And then if the user wants to delete it they can use /setdefault followed by /none.